src/libraries/System.Collections.Concurrent/src/System/ThrowHelper.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System
+{
+    internal static class ThrowHelper
+    {
+        [DoesNotReturn]
+        internal static void ThrowKeyNotFoundException(object key) => throw new KeyNotFoundException(SR.Format(SR.Arg_KeyNotFoundWithKey, key.ToString()));
```

Comment:
Make this method generic?

The method itself is cold / exceptional, but the boxing can be moved out of the call-site, thus reducing code-size there.

(Potential) downside is more generic instantiations. Maybe it's possible to incorporate this within the dictionary itself (where the generic instantiation is already done).

[rough sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8lAbhvoYGUALbKAA4AZbMAB0AJQCuAOwwBLfDHbUOAZiakGAYQYBvGgyNMNxFAwAqMXBgD6AIQgJ5MgOYAKFxgbyAlAwBeAD5LHigIAHcAaRgATwA5CAwAMQhZABMAUQQwGAEFCBlPXxVjEyZzKxtbAHEYGRgoeTBPOR9/YNDw6LjElLSZLJy8gplyYtLjQ2MvRplsABsmciQKrsiYhKTUjOzc/PlC9whgACsYMG8Aazj/abKjTowwyIYGiIZNvp3BvZHDoo3WJiCwQLgYJpudy+Er3BhwgDaACl5Bg6g0mi0MLEBDAIAAzVoYGEAXThsyg8yWzFWZnWPS2/V2wwOhXIAB4LJsgu4uXEGEC7tQHsYni8Pu9Pr1tgMhvtRu4gSCwRCXB4YSoAL5AA)

Reply:
I can move it back.